File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/Call.java
package com.roxiemobile.networkingapi.network.rest;

import com.roxiemobile.networkingapi.network.rest.request.RequestEntity;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public interface Call<T> {

    /**
     * TODO
     */
    @Nullable String tag();

    /**
     * The original request entity.
     */
    @NotNull RequestEntity<T> requestEntity();
}

File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/Call.kt
package com.roxiemobile.networkingapi.network.rest

import com.roxiemobile.networkingapi.network.rest.request.RequestEntity

interface Call<T> {

// MARK: - Properties

    /**
     * TODO
     */
    val tag: String?

    /**
     * The original request entity.
     */
    val requestEntity: RequestEntity<T>
}
